tiledb/sm/enums/filter_type.h

```diff
@@ -146,20 +145,19 @@ inline Status filter_type_enum(
   return Status::Ok();
 }
 
-/** Throws error if the input Filtertype enum is not between 0 and 16. */
+/** Throws error if the input Filtertype enum is not between 0 and 19. */
 inline void ensure_filtertype_is_valid(uint8_t type) {
-  if (type > 18) {
+  if (type > 19) {
```

Comment:
I definitely don't love seeing an un-named constant here, it will be tricky to remember this if another filter type is added.

Maybe we can add a variant to the `enum class FilterType` which is `INTERNAL_FILTER_COUNT` or something?  This definitely works with enums which don't have values assigned, but maybe it will work here anyway.

Reply:
Another way to ensure that this does not break in the future would be to claim that the type is invalid if and only if `filter_type_str` returns empty string.  That buys some robustness against someone forgetting to update *both* of those functions.

Reply:
Let's add a quick test here which just checks `if (is_filter_type_valid(i)) { CHECK(filter_type_str(i) != constants::empty_string); })` and also the other direction.  I think that's a good enough way of adding future robustness.